event-dispatcher/src/main/java/io/sk8s/event/dispatcher/JobLauncher.java

```diff
@@ -104,23 +115,17 @@ public void dispatch(String payload, Map<String, Object> headers, FunctionResour
 		return resolved;
 	}
 
-	private EnvVar[] buildEnvVars(List<Env> envList, String payload) {
-		Map<String, String> map = new HashMap<>();
-		if (envList != null) {
-			for (Env env : envList) {
-				String value = ("payload".equalsIgnoreCase(env.getValueFrom())) ? payload : "";
-				map.put(env.getName(), value);
-			}
-		}
-		map.put("MESSAGE", payload);
-		EnvVar[] vars = new EnvVar[map.size()];
-		int i = 0;
-		for (Map.Entry<String, String> entry : map.entrySet()) {
-			vars[i++] = new EnvVarBuilder()
-					.withName(entry.getKey())
-					.withValue(entry.getValue())
-					.build();
-		}
-		return vars;
+	private EnvVar[] buildEnvVars(List<FunctionEnvVar> envList, String payload) {
+		// Put the payload under the "MESSAGE" key and any envvar that had valueFrom() equal
```

Comment:
Is this a long term thing?
What will we do when payload could be anything, not just String?
Should we dump it to a Volume?